Polygon_mesh_processing/include/CGAL/Side_of_triangle_mesh.h

```diff
@@ -195,8 +203,21 @@ class Side_of_triangle_mesh
    */
   Bounded_side operator()(const Point& point) const
   {
+    bool is_outside = tree_built;
+    if(!is_outside){
+      is_outside = (point.x() < box.xmin()
+                    || point.x() > box.xmax()
+                    || point.y() < box.ymin()
+                    || point.y() > box.ymax()
+                    || point.z() < box.zmin()
+                    || point.z() > box.zmax());
+      if(is_outside)
+        return CGAL::ON_UNBOUNDED_SIDE;
+    }
+
+    tree_built = true;
```

Comment:
the Boolean seems useless.

Reply:
It enables/disables the first test, I don't see how it is useless.

Reply:
@sloriot suggest that it could be:
```c++
if(!tree_built) {
  if(point.x() < box.xmin()
     || point.x() > box.xmax()
     || point.y() < box.ymin()
     || point.y() > box.ymax()
     || point.z() < box.zmin()
     || point.z() > box.zmax()) return CGAL::ON_UNBOUNDED_SIDE;
}
tree_built = true;
```

Reply:
Ok from where he put his comment I thought he was refering to tree_built

Reply:
Well what I was saying was not to refactor the check but that the bool was never used because you don't built the tree lazily for now (AFAIS)